### PR TITLE
Fix #159 by resetting creation options to NULL

### LINKED ISSUE #159: Creation options auto passing empty array breaks Boolean Filter
With the change of #127 empty creation options are now passed as empty arrays rather than NULL to the constructors of services.
This results in a breaking change affecting the Zend\Filter\Boolean.

https://github.com/zendframework/zend-filter/blob/master/src/Boolean.php#L64

Before the PR, NULL would have been passed to the constructor by default, defaulting the "type" value to TYPE_PHP which would do regular PHP-style casts. Now that an empty array is passed instead, it assumes 0 as the type and keeps it there, because there is no 'type' override in the "creation options".

AKA this gets executed:
https://github.com/zendframework/zend-filter/blob/master/src/Boolean.php#L73
https://github.com/zendframework/zend-filter/blob/master/src/Boolean.php#L95
https://github.com/zendframework/zend-filter/blob/master/src/Boolean.php#L104
https://github.com/zendframework/zend-filter/blob/master/src/Boolean.php#L117

Setting the 'type' to 0, and as such _everything_ is casted to TRUE due to 'casting' being TRUE by default

https://github.com/zendframework/zend-filter/blob/master/src/Boolean.php#L289

So even e.g. NULL gets filtered to TRUE

EDIT: This issue was FIXED in this PR #164 and was released in version 2.7.8

### PR DESCRIPTION
**Pull request by amcsi**
I'm afraid you might consider this a BC break because of changing the signature of `setCreationOptions()`

See #159

### CODE REVIEW

File: test/AbstractPluginManagerTest.php [hunk 1]
```diff
@@ -151,6 +151,8 @@ public function testInvokableFactoryOptionsAffectMultipleInstantiations()
         $plugin2 = $pluginManager->get(Baz::class);
 
         $this->assertNotEquals($plugin1, $plugin2);
+        $this->assertSame($creationOptions, $plugin1->getOptions());
+        $this->assertNull($plugin2->getOptions());
```
Comment (Ocramius) [inline test/AbstractPluginManagerTest.php]: You need a separate test case for setting creation options once to an array, then back to `null`.

Comment (amcsi) [inline test/AbstractPluginManagerTest.php]: Can it be part of this same test method?

Comment (Ocramius) [inline test/AbstractPluginManagerTest.php]: Just add a new method with a matching name

Comment (amcsi) [inline test/AbstractPluginManagerTest.php]: @Ocramius all right, done

File: test/TestAsset/Baz.php [hunk 2]
```diff
@@ -17,4 +17,9 @@ public function __construct($options = null)
     {
         $this->options = $options;
     }
+
+    public function getOptions()
```
Comment (weierophinney) [inline test/TestAsset/Baz.php]: This isn't technically needed, as `$options` is public. As such, the tests could just look at `$instance->options`. I can make that change during merge, if necessary.

File: src/Factory/InvokableFactory.php [hunk 3]
```diff
@@ -107,7 +107,7 @@ public function createService(ServiceLocatorInterface $serviceLocator, $canonica
     /**
      * {@inheritdoc}
      */
-    public function setCreationOptions(array $creationOptions)
+    public function setCreationOptions(array $creationOptions = null)
```
Comment (weierophinney) [inline src/Factory/InvokableFactory.php]: My first inclination on reading this is that it's a BC break and/or incompatible with existing interfaces. However, on more thorough review of the code, I find this is not the case:

- `InvokableFactory` does not implement `MutableCreationOptionsInterface` directly, but instead duck-types it. As such, we can change the signature here without worries of breakage due to inconsistent implementation.
- Further, `InvokableFactory` is marked `final`, which means we do not need to worry about extending classes. This also means that the changes in `AbstractPluginManager` will not have ramifications beyond this class.

As such, I'm going to clear the "BC Break" label, as there is none at this point.
